src/redux/slices/auth.js

```diff
@@ -8,6 +24,53 @@ const initialState = {
 const authSlice = createSlice({
     name: 'auth',
     initialState,
+    reducers: {
+        logout: (state) => {
+            state.data = null;
+        }
+    },
+    extraReducers: {
+        [fetchAuth.pending]: (state) => {
+            state.status = 'loading';
+            state.data = null;
+        },
+        [fetchAuth.fulfilled]: (state, action) => {
+            state.status = 'loaded';
+            state.data = action.payload;
+        },
+        [fetchAuth.rejected]: (state) => {
+            state.status = 'error';
+            state.data = null;
+        },
+        [fetchAuthMe.pending]: (state) => {
+            state.status = 'loading';
+            state.data = null;
+        },
+        [fetchAuthMe.fulfilled]: (state, action) => {
+            state.status = 'loaded';
+            state.data = action.payload;
+        },
+        [fetchAuthMe.rejected]: (state) => {
+            state.status = 'error';
+            state.data = null;
+        },
+        [fetchRegister.pending]: (state) => {
+            state.status = 'loading';
+            state.data = null;
+        },
+        [fetchRegister.fulfilled]: (state, action) => {
+            state.status = 'loaded';
+            state.data = action.payload;
+        },
+        [fetchRegister.rejected]: (state) => {
+            state.status = 'error';
+            state.data = null;
+        },
+    }
```

Comment:
These changes apply the `addCase` function instead of using separate pointers for each action. This makes it easier to add handlers to `extraReducers` and makes the code more readable and compact.
```suggestion
    extraReducers: (builder) => {
      builder
        .addCase(fetchAuth.pending, (state) => {
          state.status = 'loading';
          state.data = null;
        })
        .addCase(fetchAuth.fulfilled, (state, action) => {
          state.status = 'loaded';
          state.data = action.payload;
        })
        .addCase(fetchAuth.rejected, (state) => {
          state.status = 'error';
          state.data = null;
        })
        .addCase(fetchAuthMe.pending, (state) => {
          state.status = 'loading';
          state.data = null;
        })
        .addCase(fetchAuthMe.fulfilled, (state, action) => {
          state.status = 'loaded';
          state.data = action.payload;
        })
        .addCase(fetchAuthMe.rejected, (state) => {
          state.status = 'error';
          state.data = null;
        })
        .addCase(fetchRegister.pending, (state) => {
          state.status = 'loading';
          state.data = null;
        })
        .addCase(fetchRegister.fulfilled, (state, action) => {
          state.status = 'loaded';
          state.data = action.payload;
        })
        .addCase(fetchRegister.rejected, (state) => {
          state.status = 'error';
          state.data = null;
        });
    },

```